DQMOffline/L1Trigger/src/L1TTauOffline.cc

```diff
@@ -533,10 +533,23 @@ bool L1TTauOffline::matchHlt(edm::Handle<trigger::TriggerEvent> const& triggerEv
 
   trigger::TriggerObjectCollection trigObjs = triggerEvent->getObjects();
 
-  for (auto trigIndexIt = m_trigIndices.begin(); trigIndexIt != m_trigIndices.end(); ++trigIndexIt) {
-    const vector<string> moduleLabels(m_hltConfig.moduleLabels(*trigIndexIt));
-    const unsigned moduleIndex = m_hltConfig.size((*trigIndexIt)) - 2;
+  vector<int>::const_iterator trigIndexIt = m_trigIndices.begin();
+  vector<int>::const_iterator trigIndexEnd = m_trigIndices.end();
 
+  for (; trigIndexIt != trigIndexEnd; ++trigIndexIt) {
+    const vector<string> moduleLabels(m_hltConfig.moduleLabels(*trigIndexIt));
+    // V.M. 2023.03.08 Same issue as in the L1TMuonDQMOffline.cc - some modules are behind hltBoolEnd, but we want the last one before the "hltBoolEnd"
+    unsigned int moduleIndex = 999999;
+    for (int ii = 0; ii < int(moduleLabels.size()); ii++) {
+      if (moduleLabels[ii] == "hltBoolEnd") {
+        moduleIndex = ii - 1;
+        break;
+      }
+    }
+    if (moduleIndex == 999999) {
+      edm::LogError("L1TMuonDQMOffline") << "Found no module label in trigger " << (*trigIndexIt) << endl;
+      continue;
```

Comment:
Could you not just use an iterator on the `moduleLabels` vector here and simply check if has hit the end of the vector? It may be technically equivalent, but it might be a little easier to read and more robust than simply checking against a large int.

Reply:
Sorry, maybe I misunderstood the question, but the approach with the iterator will work if the "hltBoolEnd" is not the last module, but if it is and we use the iterator == end of vector, we'll drop the event.

Reply:
Hi @aloeliger, is the solution proposed by @perrotta ok with you too? That one is implemented with the latest commit

Reply:
Of course, he's a better reference than I.